packages/ui/src/components/ValidatorList/KeystoresDataGrid.tsx

```diff
@@ -61,6 +69,106 @@ export default function KeystoresDataGrid({
     openDashboardTab();
   }, [validatorSummaryURL]);
 
+  useState<CustomValidatorUpdateRequest>();
+  const [validatorsSubscriptionMap, setValidatorsData] =
+    useState<SmoothStatusByPubkey>({});
+  const [oracleCallError, setOracleCallError] = useState<string>();
+
+  // Check that Smooth API returns an expected response format
+  function isValidOracleResponse(response: SmoothValidatorByIndexApiResponse): boolean {
+    return (
+      response &&
+      (Array.isArray(response.found_validators) ||
+        response.found_validators === null) &&
+      (Array.isArray(response.not_found_validators) ||
+        response.not_found_validators === null)
+    );
+  }
+
+  // This function fetches the subscription status of the validators in the rows array
+  // It uses the validator index to fetch the status from the Oracle API
+  // It updates the validatorsSubscriptionMap state with the new data
+  const fetchValidatorsData = async (rows: CustomValidatorGetResponse[]) => {
+    const newValidatorSubscriptionStatus: SmoothStatusByPubkey = {};
+    // Filter rows to include only those with an index
+    const rowsWithIndex = rows.filter(row => row.index);
+  
+    // Only proceed if there is at least one row with an index
+    // new validatos will not have an index yet
+    if (rowsWithIndex.length > 0) {
```

Comment:
You could remove this and include inside the try-catch:

```
if (rowsWithIndex.length < 1 ) throw ... (error message)
```

I suggest this because you already handle setting the oracle error on the catch. This way you could avoid this duplication:
![image](https://github.com/dappnode/StakingBrain/assets/144998261/196a38fa-3763-43ee-954d-64dee9a699b1)

Reply:
fixed